core/primitives/rle_bitset/rle_bitset.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_PRIMITIVES_RLE_BITSET_RLE_BITSET_HPP
+#define CPP_FILECOIN_CORE_PRIMITIVES_RLE_BITSET_RLE_BITSET_HPP
+
+#include <set>
+
+#include "codec/rle/rle_plus.hpp"
+#include "common/outcome_throw.hpp"
+
+namespace fc::primitives {
+  struct RleBitset : public std::set<uint64_t> {
+    using set::set;
+
+    inline RleBitset(set &&s) : set{s} {}
+  };
+
+  template <class Stream,
+            typename = std::enable_if_t<
+                std::remove_reference_t<Stream>::is_cbor_encoder_stream>>
+  Stream &operator<<(Stream &&s, const RleBitset &set) {
+    return s << codec::rle::encode(set);
+  }
+
+  template <class Stream,
+            typename = std::enable_if_t<
+                std::remove_reference_t<Stream>::is_cbor_decoder_stream>>
+  Stream &operator>>(Stream &&s, RleBitset &set) {
+    std::vector<uint8_t> rle;
+    s >> rle;
+    auto decoded = codec::rle::decode<RleBitset::value_type>(rle);
```

Comment:
we've already made some number of convenient macro definitions like OUTCOME_TRY(), EXPECT_OUTCOME_TRUE/FALSE. Something like OUTCOME_EXCEPT() implementing this behavior would be useful. There are places like this where we need to return outcome error in form of exception.